### PR TITLE
S04US17

### PR DESCRIPTION
**Pull request by Chikoulat**
FEAT: table user and message created. 
FEAT: joined the tables. 
UPDATE: updated candidate component, recovered data from backend.

### CODE REVIEW

File: backend/database/schema.sql [hunk 1]
```diff
@@ -36,6 +38,7 @@ CREATE TABLE company (
     establishment_date DATE  NOT NULL,
     company_sector_id INT  NOT NULL,
     user_id INT NOT NULL,
+    user_id INT NOT NULL,
```
Comment (YOUNS28100) [inline backend/database/schema.sql:41]: duplicate

File: backend/database/schema.sql [hunk 2]
```diff
@@ -124,6 +127,7 @@ CREATE TABLE user (
     email_notification_active BOOL  NOT NULL,
     image VARCHAR(255)  NULL,
     registration_date DATETIME NOT NULL DEFAULT NOW(),
+    registration_date DATETIME NOT NULL DEFAULT NOW(),
```
Comment (YOUNS28100) [inline backend/database/schema.sql:130]: duplicate

**Review by StalkerVince — APPROVED** [review]

File: backend/seed.js [hunk 3]
```diff
@@ -180,6 +181,7 @@ const seed = async () => {
             fakerFR.datatype.boolean(0.6),
             fakerFR.datatype.boolean(0.9),
             fakerFR.internet.avatar(),
+            fakerFR.date.recent({ days: 1 }),
```
Comment (YOUNS28100) [inline backend/seed.js:184]: 9 dependencies for 8 attributes

File: backend/seed.js [hunk 4]
```diff
@@ -167,7 +168,7 @@ const seed = async () => {
             fakerFR.word.adjective(),
             faker.helpers.arrayElement(skillLevel, 1),
             fakerFR.number.int({ min: 1, max: 20 }),
-            fakerFR.number.int({ min: 1, max: 20 }),
+            fakerFR.number.int({ min: 1, max: 200 }),
```
Comment (YOUNS28100) [inline backend/seed.js:171]: 1557 job created in loop, set max 1557

File: backend/seed.js [hunk 5]
```diff
@@ -94,7 +94,7 @@ const seed = async () => {
           ]
         ),
         database.query(
-          "INSERT INTO company (name, image, description, website, establishment_date, company_sector_id, user_id) VALUES (?, ?, ?, ?, ?, ?, ?)",
+          "INSERT INTO company (name, image, description, website, establishment_date, company_sector_id, user_id, user_id) VALUES (?, ?, ?, ?, ?, ?, ?,?)",
```
Comment (YOUNS28100) [inline backend/seed.js:97]: duplicate

File: backend/seed.js [hunk 6]
```diff
@@ -77,7 +77,7 @@ const seed = async () => {
           ]
         ),
         database.query(
-          "insert into candidate (firstname, lastname, date_of_birth, wanted_salary, user_id) VALUES (?, ?, ?, ?, ?)",
+          "insert into candidate (first_name, last_name, date_of_birth, wanted_salary, user_account_id) VALUES (?, ?, ?, ?, ?)",
```
Comment (YOUNS28100) [inline backend/seed.js:80]: underscore in firstname/lastname ?

File: backend/database/schema.sql [hunk 7]
```diff
@@ -79,7 +82,7 @@ CREATE TABLE job (
     type VARCHAR(30)  NOT NULL,
     description TEXT  NOT NULL,
     hours_worked INT NOT NULL,
-    created_date DATETIME NOT NULL DEFAULT NOW(),
+    created_date DATETIME NOT NULL DEFAULT NOW() DEFAULT NOW(),
```
Comment (YOUNS28100) [inline backend/database/schema.sql:85]: duplicate

File: backend/database/schemaFK.sql
Comment (YOUNS28100) [inline backend/database/schemaFK.sql:1]: we need to see again this file

File: backend/seed.js (hunk 6 above)
Comment (YOUNS28100) [inline backend/seed.js:80]: user_id, not user_account_id

File: backend/seed.js [hunk 8]
```diff
@@ -107,6 +107,7 @@ const seed = async () => {
             fakerFR.date.past({ years: 30 }),
             fakerFR.number.int({ min: 1, max: 20 }),
             fakerFR.number.int({ min: 1, max: 20 }),
+            fakerFR.number.int({ min: 1, max: 20 }),
```
Comment (YOUNS28100) [inline backend/seed.js:110]: duplicate to delete

**Review by YOUNS28100 — CHANGES_REQUESTED** [review]

**Review by sdiakhate — APPROVED** [review]

File: backend/seed.js (hunk 4 above)
Comment (Chikoulat) [reply backend/seed.js:171]: we don't need to have 1557 sector, a sector can take many job offers

**Pull request closed without merging**